### PR TITLE
Create sign up mutation

### PR DESCRIPTION
**Pull request by 00dav00**
- [x] Add signup mutation
- [x] Add url helper to create inline mutation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@
 /*.sqlite3
 /spec/dummy/db/development.sqlite3
 /spec/dummy/db/test.sqlite3
+/graphql_devise-0.1.0.gem
```
Comment (mcelicalderon) [inline .gitignore]: Tus will change with every version use *.gem instead

File: app/graphql/graphql_devise/mutations/sign_up.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    class SignUp < Base
+      argument :email,                 String, required: true
+      argument :name,                  String, required: false
+      argument :password,              String, required: true
+      argument :password_confirmation, String, required: true
+      argument :confirm_success_url,   String, required: false
+      argument :config_name,           String, required: false
+
+      def resolve(email:, **attrs)
+        redirect_url = attrs.delete(:confirm_success_url)
```
Comment (mcelicalderon) [inline app/graphql/graphql_devise/mutations/sign_up.rb]: I would add kargs that are required directly in the resolve params so you don't have to do something like this

File: app/graphql/graphql_devise/mutations/sign_up.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
+module GraphqlDevise
+  module Mutations
+    class SignUp < Base
+      argument :email,                 String, required: true
+      argument :name,                  String, required: false
```
Comment (mcelicalderon) [inline app/graphql/graphql_devise/mutations/sign_up.rb]: Name is an attribute only on the dummy app, we need a way to configure that, or at least remove this one for now. You can use a custom mutation in the dummy app that extends from this one

File: app/graphql/graphql_devise/mutations/sign_up.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        end
+      end
+
+      protected
+
+      def confirmable_enabled?(resource)
+        resource.respond_to?(:confirmed_at)
+      end
+
+      def active_for_authentication?(resource)
```
Comment (mcelicalderon) [inline app/graphql/graphql_devise/mutations/sign_up.rb]: This one seems redundant when you can call that directly on resource

File: app/graphql/graphql_devise/mutations/sign_up.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        resource.active_for_authentication?
+      end
+
+      def provider
+        :email
+      end
+
+      # NOTE: Devise controller method, find a way to re use it
```
Comment (mcelicalderon) [inline app/graphql/graphql_devise/mutations/sign_up.rb]: You could put it on the base mutation if some other mutation uses it

File: app/helpers/graphql_devise/mailer_helper.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+module GraphqlDevise
+  module MailerHelper
+    extend ActiveSupport::Concern
```
Comment (mcelicalderon) [inline app/helpers/graphql_devise/mailer_helper.rb]: Not necessary, right?

File: app/helpers/graphql_devise/mailer_helper.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    protected
```
Comment (mcelicalderon) [inline app/helpers/graphql_devise/mailer_helper.rb]: I think helpers are usually just public

File: lib/graphql_devise/rails/routes.rb [hunk 8]
```diff
@@ -21,6 +19,7 @@ def mount_graphql_devise_for(resource, opts = {})
       default_mutations = {
         login:           GraphqlDevise::Mutations::Login,
         logout:          GraphqlDevise::Mutations::Logout,
+        signUp:          GraphqlDevise::Mutations::SignUp,
```
Comment (mcelicalderon) [inline lib/graphql_devise/rails/routes.rb]: This should be snake_case

File: lib/graphql_devise/rails/routes.rb [hunk 9]
```diff
@@ -35,9 +34,11 @@ def mount_graphql_devise_for(resource, opts = {})
           new_mutation
         end
 
-        GraphqlDevise::Types::MutationType.field("#{mapping_name}_#{action}", mutation: used_mutation)
+        GraphqlDevise::Types::MutationType.field("#{mapping_name}_#{action.to_s.underscore}", mutation: used_mutation)
```
Comment (mcelicalderon) [inline lib/graphql_devise/rails/routes.rb]: Use snake case everywhere until it is required otherwise

File: lib/graphql_devise/rails/routes.rb [hunk 10]
```diff
@@ -35,9 +34,11 @@ def mount_graphql_devise_for(resource, opts = {})
...
       end
 
+      Devise.mailer.send(:add_template_helper, GraphqlDevise::MailerHelper)
```
Comment (mcelicalderon) [inline lib/graphql_devise/rails/routes.rb]: There might be an engine way not to need this, but is the method not public?

File: lib/graphql_devise/rails/routes.rb (hunk 9 above)
Comment (00dav00) [reply lib/graphql_devise/rails/routes.rb]: Naming here uses snake case, note `"#{mapping_name}_`

File: lib/graphql_devise/rails/routes.rb (hunk 10 above)
Comment (00dav00) [reply lib/graphql_devise/rails/routes.rb]: Let me take a look, but it is private [doc](https://apidock.com/rails/AbstractController/Helpers/ClassMethods/add_template_helper)

File: lib/graphql_devise/rails/routes.rb (hunk 9 above)
Comment (mcelicalderon) [reply lib/graphql_devise/rails/routes.rb]: Yes, calling underscore here is not necessary as you changed line 22

**Pull request merged**